src/bindings/c/include/openvino/c/ov_infer_request.h

```diff
@@ -70,68 +71,207 @@ OPENVINO_C_API(ov_status_e)
  * @ingroup infer_request
  * @param infer_request A pointer to the ov_infer_request_t.
- * @param idx Index of the input tensor. If @p idx is greater than the number of model inputs, an exception is thrown.
+ * @param port Port of the input or output tensor, which can be got by calling ov_model_t/ov_compiled_model_t interface.
  * @param tensor Reference to the tensor.
  * @return Status code of the operation: OK(0) for success.
  */
 OPENVINO_C_API(ov_status_e)
-ov_infer_request_set_input_tensor(ov_infer_request_t* infer_request, size_t idx, const ov_tensor_t* tensor);
+ov_infer_request_set_tensor_by_port(ov_infer_request_t* infer_request,
+                                    const ov_output_node_t* port,
+                                    const ov_tensor_t* tensor);
 
 /**
- * @brief Gets an input/output tensor to infer on.
+ * @brief Set an input/output tensor to infer request for the port.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param port Const port of the input or output tensor, which can be got by call interface from
+ * ov_model_t/ov_compiled_model_t.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_tensor_by_const_port(ov_infer_request_t* infer_request,
+                                          const ov_output_const_node_t* port,
+                                          const ov_tensor_t* tensor);
+
+/**
+ * @brief Set an input tensor to infer on by the index of tensor.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param idx Index of the input port. If @p idx is greater than the number of model inputs, an error will return.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_input_tensor_by_index(ov_infer_request_t* infer_request,
+                                           const size_t idx,
+                                           const ov_tensor_t* tensor);
+
+/**
+ * @brief Set an input tensor for the model with single input to infer on.
+ * @note If model has several inputs, an error will return.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_input_tensor(ov_infer_request_t* infer_request, const ov_tensor_t* tensor);
+
+/**
+ * @brief Set an output tensor to infer by the index of output tensor.
+ * @note Index of the output preserved accross ov_model_t, ov_compiled_model_t.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param idx Index of the output tensor.
+ * @param tensor Reference to the tensor.
```

Comment:
But I see reference in a lot of places. I think it can be fixed separately in next PR.

Reply:
There is some inappropriate description in documentation, it is input argument and pass by pointer.